src/ctr_logging.c

```diff
@@ -606,6 +614,9 @@ static void reopen_k8s_file(void)
 	/* Close the current k8s_log_fd */
 	close(k8s_log_fd);
 
+	/* Open with O_TRUNC: reset bytes written */
+	k8s_bytes_written = 0;
```

Comment:
wouldn't we want to reset *total_* as well here? Otherwise, the file rotation will rotate but won't be written to again

Reply:
The current code doesn't so I didn't think very hard about it -- from my understanding of the current code, `total_` indeed isn't reset, so logging stops as soon as the container has output log_global_size_max bytes (this patch makes it "minus the original file size" as I thought using just log global size max alone ought to limit the log size as well without rotating)
in other words:
 - log_size_max is used to cap the file size, rotating as necessary
 - log_global_size_max either caps the file size if used alone or limits the number of times the file is rotated

Looking at the commit that introduced it - 98058b14d9b1a2d6ecaaf0f26e4c1dde5e32865a "add log-global-size-max option to limit the total output conmon processes" - it might make more sense to not init k8s_total_bytes_written to the original file size, but then a user with only that option would risk the file growing indefinitely when container restarts so I'm not sure that's intended?
Either way we probably shouldn't just reset the total bytes written, if we do both vars would just be identical all the time and could be folded back together.

Reply:
ah sorry it's been a while since I've looked here, I forgot there were two cli flags for each situation.